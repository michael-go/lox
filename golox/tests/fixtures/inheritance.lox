class Doughnut {
  init() {
    print "init Doughnut";
  }

  cook() {
    print "Fry until golden brown.";
  }
}

class BostonCream < Doughnut {
  init() {
    super.init();
    print "init BostonCream";
  }

  cook() {
    super.cook();
    print "Pipe full of custard and coat with chocolate.";
  }
}

BostonCream().cook();