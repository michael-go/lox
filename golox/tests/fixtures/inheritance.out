# exit code: 0
# stdout:
Fry until golden brown.
Pipe full of custard and coat with chocolate.
